partiql-parser/src/peg.rs

```diff
@@ -64,8 +67,12 @@ impl<'val, R: RuleType> PairExt<'val, R> for Pair<'val, R> {
         self.as_span().end_pos().line_col().try_into()
     }
 
-    fn syntax_error<T>(&self) -> ParserResult<T> {
-        syntax_error(format!("Unexpected rule: {:?}", self), self.start()?.into())
+    fn unexpected<T>(&self) -> ParserResult<T> {
+        self.syntax_error(format!("Unexpected rule: {:?}", self))
+    }
+
+    fn syntax_error<T, S: Into<String>>(&self, message: S) -> ParserResult<T> {
+        syntax_error(message, self.start()?.into())
```

Comment:
If `self.start()` returns `Err`, that's what will bubble up instead of the actual syntax error. Is that possible/likely? At a glance I'd think we would want something like:
```rust
// Handle the bad location case separately somehow
let position = self.start().ok_or_else(|| Position::Unknown).into();
// Always return the result from `syntax_error`
syntax_error(message, position)
```

Reply:
This is definitely a better mapping of a bad position--will change.